bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/SamsungTvAppWatchService.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.samsungtv.internal;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchEvent.Kind;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.samsungtv.internal.protocol.RemoteControllerWebSocket;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.service.AbstractWatchService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SamsungTvAppWatchService} provides a list of apps for >2020 Samsung TV's
+ * File should be in json format
+ *
+ * @author Nick Waterton - Initial contribution
+ */
+@NonNullByDefault
+public class SamsungTvAppWatchService extends AbstractWatchService {
+    private static final String APPS_PATH = OpenHAB.getConfigFolder() + File.separator + "misc";
+    private static final String APPS_FILE = "samsungtv.applist";
+
+    private final Logger logger = LoggerFactory.getLogger(SamsungTvAppWatchService.class);
+    private final RemoteControllerWebSocket remoteControllerWebSocket;
+    private String host = "Unknown";
+    private boolean started = false;
+
+    public SamsungTvAppWatchService(String host, RemoteControllerWebSocket remoteControllerWebSocket) {
+        super(APPS_PATH);
+        this.host = host;
+        this.remoteControllerWebSocket = remoteControllerWebSocket;
+    }
+
+    public void start() {
+        logger.info("{}: Starting Apps File monitoring service", host);
+        started = true;
+        makeFileDir();
+        readFileApps();
+        activate();
+    }
+
+    public boolean getStarted() {
+        return started;
+    }
+
+    /**
+     * Create directory and file if they don't exist
+     *
+     */
+    public void makeFileDir() {
```

Comment:
On some systems this directory already exists, as the exec binding creates it. If you don't use the exec binding, it doesn't exist. Some people were confused that they had to manually create a directory, and could create it with the wrong permissions. It seemed easier to automatically create it it it wasn't already there.

It only gets created if the TV does not support the app API. It isn't always created.

Reply:
On the other hand some users could get confused by some strange directory and empty file suddenly existing on their system. I have $OPENHAB_CONF under Git version control, so I would certainly notice and be annoyed. And I wouldn't be able to just clean it up again - the binding would then create it one more time. The README is already quite verbose, so it could also explain this - including the importance of having the correct permissions.